LM-50: modify services and controllers to handle new project model

rgillen-uno opened the pull request:
Fixing the services, controllers and tests that were broken by the Model changes introduced by [LM-13-1](https://github.com/UnosquareBelfast/admin-backend/pull/82).

rgillen-uno committed:
LM-13-1: Implement the project schema and DAL model changes required.

rgillen-uno committed:
LM-13-1: Add to migration to ensure default project created for every team/client.

rgillen-uno committed:
LM-13-1: Remove changes accidentally committed.

rgillen-uno committed:
LM-13-1: Amend comments on migration

rgillen-uno committed:
LM-13-2: Add extra fields to Project and refactor ContractService.

rgillen-uno committed:
LM-13-1: Add dockerfile to build custom postgres container with postgres-contrib

rgillen-uno committed:
Revert "LM-13-1: Add dockerfile to build custom postgres container with postgres-contrib"

This reverts commit f09ea59053cb608709501908bbf0ce6eaa8cd60f.

rgillen-uno committed:
LM-13-2: Project self-referencing and fixes for services.
- Added new test mocks for several repos.
- Modified ContractService tests.
- Modified Contract and Team DTOs and related mappings between DTO and model.

rgillen-uno committed:
Merge branch 'LM-13-1-implement-project-schema-model-changes' into LM-13-2-modify-services-and-controllers-to-handle-new-project-model

rgillen-uno committed:
LM-13-2: Fix all tests and define Contract=>ContractDTO mapping to correctly map ProjectName and CLientName.

rgillen-uno committed:
LM-13-2: Add nullable foreign key int to Project model

rgillen-uno committed:
LM-13-2: Fix mapping issues with Contract

rgillen-uno committed:
LM-50: Added test for Contract Mapper

rgillen-uno committed:
LM-50: Refactor lambda expression variable names to be more meaningful.

kkuczkowski-unosq commented on AdminCore.Services.Tests/DashboardServiceLogicTests.cs:
```diff
@@ -18,10 +18,10 @@ public DashboardServiceLogicTests()
       var resultList = _mockDatabase.EventRepository.Where(evnt => DashboardService.EmployeeDashboardEventsQuery(1, new DateTime(2018, 12, 5), evnt)).ToList();
       Assert.Single(resultList);
-      Assert.Equal((int)EventStatuses.Approved, resultList.First().EventStatusId);
+      Assert.Equal((int)EventStatuses.AwaitingApproval, resultList.First().EventStatusId);
```
reason for this change is unclear, why was this changed?

rgillen-uno replied:
The eventRepo didn't contain an event that was Approved with these particular parameters, so the test was failing. So I changed the assertion to assert for an event that did exist in the mock db.

kkuczkowski-unosq commented on AdminCore.Services.Tests/MappingProfileTests/ContractMappingProfileTests.cs:
```diff
@@ -0,0 +1,40 @@
+        {
+            _mapper = new Mapper(new MapperConfiguration(cfg =>
+            {
+//                cfg.ValidateInlineMaps = false;
```
clean up?

rgillen-uno replied:
Cleaned comments.

kkuczkowski-unosq reviewed (approved):
looks good to me :shipit:

rgillen-uno committed:
LM-50: Removed comments.

rgillen-uno committed:
lM-50: Add Project Repository to IDatabaseContext.

BrownMark commented on AdminCore.DatabaseMigration/db/migrations/V1.11__add_project.sql:
```diff
@@ -0,0 +1,83 @@
+  project
+WHERE
+  team.client_id = project.client_id;
+
```
Make project_id not nullable.

rgillen-uno replied:
Done

rgillen-uno committed:
LM-50: Add generic mock repository setup method to reduce code duplication.

rgillen-uno committed:
LM-53: implement changes to team controller service dto viewmodel (#85)

* LM-53: Add new GetTeamsByProjectId to controller/service and add initial tests.

* LM-53: Add some new tests for GetTeamByProjectId

* LM-53: Refactor Team service tests to reduce code duplication.

* LM-55: Add new contract controller/service method and initial tests.

* LM-53: Remove comments.

* LM-55: Fix initial test for contract service.

* LM-53: Add tests for new controller method.

* LM-55: Add ContractController test for GetContractByProjectId.

* LM-53: Refactor tests to reduce code duplication.

* LM-53: Change two tests to use mocked mappers.

* LM-55: Fix mapper references.

* LM-53: Fix team controller test mapper for test to return correct instance.

* LM-55: Add correct ProducesResponseType attributes to ContractController.

* LM-53: Update TeamController tests to match format of later PRs.

* LM-53: Rework how tests ORM and Mappers are used for new Team tests.

* LM-55: Rework contract controller and service tests.

* LM-53: Update Team Service tests to not mock the repository.

* LM-55: Modify contract tests to not mock repository.

* LM-53: Modify mocked db and orm setup for base class.

* LM-55: Update tests.

* LM-56: implement changes to project controller service dto viewmodel (#88)

* amend

* LM-56: Add project viewmodels and controller.

* LM-56: Modify viewmodels and add db context save statements.

* LM-56: Update project controller endpoints

* LM-56: Update project controller error handling.

* LM-56: Add initial project controller tests.

* LM-56: Update Project controller class data to support GetProjectsById tests.

* LM-56: Add more tests for ProjectController and refactor method name.

* LM-56: Add more ProjectController tests.

* LM-56: Refactor project service and tests.

* LM-56: Update version of NSubstitute in WebApi.Tests and fix project controller tests.

* LM-56: Add and Update project service tests.

* LM-56: Add project mapping tests.

* LM-56: Add correct ProducesResponseType types to ProjectController.

* LM-56: Rework Project Controller/Service unit tests.

* LM-56: Modify project tests to not mock repository.

* LM-56: Update project interface.

* LM-56: Add optional parameters for getProjects and remove redundant methods.

* LM-56: Add more specific spies for Project Service tests.

* LM-56: Modify controller project to use the hand bad request action.

* LM-56: Update project controller tests to use BeEmpty.

* LM-54: implement changes to dashboard controller service dto viewmodel (#89)

* LM-54: Add dashboard service/controller functionality, tests, mapping and refactor some names.
Add new Dtos and viewmodels.

* LM-54: Add new tests for DashboardService and modify GetTeamDashboardEvents.
- Add tests to handle empty/null list scenarios for GetTeamDashboardEvents of DashboardService
- Modify GetTeamDashboardEvents to handle null returned from ORM.

* LM-54: Add null handling to GetEmployeeTeamSnapshot and tests for dashboard controller.

* LM-54: Add correct ProducesResponseType attributes to DashboardController.

* LM-54: Remove null-conditionals and redundant dashboard service test.

* LM-53: Specify number of receives in team service tests.

rgillen-uno committed:
LM-50: Change Team project_id to be Not Null.

rgillen-uno committed:
Merge branch 'LM-13-add-project-structure' into LM-50-modify-services-and-controllers-to-handle-new-project-model
